float64 time_x
float32 signal_y